packages/api/lambdas/sf-event-sqs-to-db-records.js

```diff
@@ -1,22 +1,68 @@
 'use strict';
 
 const get = require('lodash/get');
+const semver = require('semver');
+
 const { parseSQSMessageBody, sendSQSMessage } = require('@cumulus/aws-client/SQS');
 const log = require('@cumulus/common/log');
-const { getMessageExecutionArn } = require('@cumulus/message/Executions');
+const {
+  getKnexClient,
+  Executions,
+  tableNames,
+  database,
+} = require('@cumulus/db');
+const {
+  getMessageExecutionArn,
+  getMessageExecutionParentArn,
+  getMessageCumulusVersion,
+} = require('@cumulus/message/Executions');
 const Execution = require('../models/executions');
 const Granule = require('../models/granules');
 const Pdr = require('../models/pdrs');
 const { getCumulusMessageFromExecutionEvent } = require('../lib/cwSfExecutionEventUtils');
 
-const saveExecutionToDb = async (cumulusMessage) => {
+const isPostRDSDeploymentExecution = (cumulusMessage) => {
+  const minimumSupportedRDSVersion = process.env.RDS_DEPLOYMENT_CUMULUS_VERSION;
+  if (!minimumSupportedRDSVersion) {
+    throw new Error('RDS_DEPLOYMENT_CUMULUS_VERSION environment variable must be set');
+  }
+  const cumulusVersion = getMessageCumulusVersion(cumulusMessage);
+  return cumulusVersion
+    ? semver.gte(cumulusVersion, minimumSupportedRDSVersion)
+    : false;
+};
+
+const shouldWriteExecutionToRDS = async (cumulusMessage, knex) => {
+  const executionIsPostDeployment = isPostRDSDeploymentExecution(cumulusMessage);
+  const parentArn = getMessageExecutionParentArn(cumulusMessage);
+  if (!executionIsPostDeployment || !parentArn) return executionIsPostDeployment;
```

Comment:
Hmm, yeah-  This works because there's no other referral that might be broken that we aren't migrating?

Reply:
Actually, executions refer to collections and async operations as well, so we should probably account for those as well. Those entities **should** be in RDS from the pre-deployment migration, but of course things can go wrong and we should account for that. Good catch.